src/bots/reddit/actions/post_actions.py

```diff
@@ -56,24 +56,31 @@ def repost(self, roll=1, subreddit=None):
       log.info("running repost")
       # log.info("running _repost")
       post = self.get_post(subreddit=subreddit)
-      log.info(f"reposting post: {post.id}")
+      p_url = post.url
+      while p_url.endswith(('.jpg', '.png', '.gif', '.jpeg')):
+        if requests.get(p_url).status_code == 404 :
```

Comment:
I think instead of checking if status_code is 404, we should check if it's 200 or not. It's better since if status_code can be 300, 500 and we still can't repost the image.

Reply:
I agree that checking for a single success value is more resilient than checking for a single failure value. there could be various failure values.